### PR TITLE
Update docs for all spelling errors, and add auth0 session and spec.presence

### PR DESCRIPTION
**Pull request by jrwpatterson**
Signed-off-by: Joe Patterson <jrwpatterson@gmail.com>

## Hey, I just made a Pull Request!

Docs update

- updates `descriptor-format.md` to add the `spec.presence` attribute documentation that was missing
- updates auth0 `provider.md` to add the need for a session for it to work
- updates vale `accept.txt` file with missing spellings to stop the errors on lint:docs 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/auth/auth0/provider.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ auth:
         audience: ${AUTH_AUTH0_AUDIENCE}
         connection: ${AUTH_AUTH0_CONNECTION}
         connectionScope: ${AUTH_AUTH0_CONNECTION_SCOPE}
+  session:
+    secret: 'supersecret'
```
Comment (Rugvip) [inline docs/auth/auth0/provider.md:41]: I think it's best to make this an env var as well, to make sure that people fill it in with their own secret. Probably `AUTH_SESSION_SECRET`

File: .github/vale/Vocab/Backstage/accept.txt [hunk 2]
```diff
@@ -1,46 +1,123 @@
+_event
```
Comment (Rugvip) [inline .github/vale/Vocab/Backstage/accept.txt:1]: What's the purpose of these changes? In general no code symbols should be needed here because Vale doesn't check things wrapped in code blocks

File: .changeset/fifty-paws-attack.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/catalog-model': patch
+'@backstage/plugin-scaffolder-common': patch
+'@backstage/plugin-search-backend-module-elasticsearch': patch
+'@backstage/plugin-search-backend-node': patch
+---
+
+Documentation update fixing the vale missing spellings, fixing errors, updating the `provider` document to include updates about auth0 needing a session, updating `descriptor-format` to include spec.presence on location which is missing
```
Comment (Rugvip) [inline .changeset/fifty-paws-attack.md:8]:
```suggestion
Fixed spelling mistakes in documentation.
```

We can keep this one simple, no need to dig into the `docs/` changes.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 🎉

File: .github/vale/Vocab/Backstage/accept.txt (hunk 2 above)
Comment (jrwpatterson) [reply .github/vale/Vocab/Backstage/accept.txt:1]: My local Vale is checking everything and I get thousands of errors

File: docs/auth/auth0/provider.md (hunk 1 above)
Comment (jrwpatterson) [reply docs/auth/auth0/provider.md:41]: Sure

**Review by Rugvip — CHANGES_REQUESTED** [review]
